### ISSUE TITLE
[BUG] Constant exceptions when working with multiple environments

### ISSUE THREAD

**Issue by kowgli** [MarkMpn/Sql4Cds#608]
**Describe the bug**
When working with queries that span multiple environments the the tool keeps throwing exceptions while writing the queries.
This is quite annoying, the message boxes appear every few typed characters and take over focus.

**To Reproduce**
Connect to more than 1 environment and write a query that join two tables from the two environments.
Note the query must be written, not copy pasted.

```sql
select *
from 
    Env1.dbo.account a1
    join Env2.dbo.account a2 on a1.accountid = a2.accountid
```

**Expected behavior**
No dialog boxes pop up. Even if the tool encounters some metadata / typing completion errors it shouldn't pop up a modal dialog box.

**Screenshots**
![image](https://github.com/user-attachments/assets/7ba5b96e-7ad6-4ab0-9510-62b6d101f7cf)

**Environment (please complete the following information):**
 - SQL 4 CDS edition: XrmToolBox
 - Results of `SELECT @@VERSION`: 
        Microsoft Dataverse - 9.2.24104.202
	SQL 4 CDS - 9.4.1.0
	nov 11 2024 10:59:24
	Copyright © 2020 - 2024 Mark Carrington

**Additional context**
Error messages:

**Message 1**
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.ServiceModel.FaultException: Unknown table msdyn_presence
   at MarkMpn.Sql4Cds.XTB.SharedMetadataCache.get_Item(String name)
   at MarkMpn.Sql4Cds.Engine.MetaMetadataCache.get_Item(String name)
   at MarkMpn.Sql4Cds.Engine.MetadataExtensions.<>c__DisplayClass8_0.<GetVirtualAttributes>b__0(String e)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Enumerable.Max(IEnumerable`1 source)
   

**Message 2**
************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at MarkMpn.Sql4Cds.Engine.MessageParameter.GetSqlDataType(DataSource dataSource)
   at MarkMpn.Sql4Cds.XTB.Autocomplete.TVFAutocompleteItem.<>c.<get_ToolTipText>b__8_3(MessageParameter p)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.String.Join(String separator, IEnumerable`1 values)
   at MarkMpn.Sql4Cds.XTB.Autocomplete.TVFAutocompleteItem.get_ToolTipText()
   at AutocompleteMenuNS.AutocompleteListView.ShowToolTip(AutocompleteItem autocompleteItem, Control control)
   at AutocompleteMenuNS.AutocompleteListView.set_SelectedItemIndex(Int32 value)
   at AutocompleteMenuNS.AutocompleteMenu.BuildAutocompleteList(Boolean forced)
   at AutocompleteMenuNS.AutocompleteMenu.ShowAutocomplete(Boolean forced)
   at AutocompleteMenuNS.AutocompleteMenu.timer_Tick(Object sender, EventArgs e)
   at System.Windows.Forms.Timer.OnTick(EventArgs e)
   at System.Windows.Forms.Timer.TimerNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by MarkMpn** [comment]
I believe this is resolved in v9.5, can you please try updating and reopen this if it still occurs?

**Comment by kowgli** [comment]
Hi @MarkMpn 

It seems to be fixed. 

Thanks a lot for this wonderful tool! For me it's the no 1 useful tool in the toolbox.
